src/oods_statement.rs

```diff
@@ -229,24 +265,21 @@ impl MainProof {
         result.push(memory_value);
 
         result.push(U256::from(pages.len()));
-        let sorted_pages: Vec<Vec<U256>> = pages.into_values().collect();
 
-        for (i, page) in sorted_pages.iter().enumerate() {
-            // Append the page's first address, size and hash
-            if i != 0 {
+        for i in 0..pages.len() {
+            let page = pages.get(&(i as u32)).unwrap();
```

Comment:
this is a bad pattern, you should do `for (idx, page) in pages.iter().enumerate() {`